pkg/crc/cluster/cluster.go

```diff
@@ -192,18 +192,50 @@ func RemovePullSecretFromCluster(ocConfig oc.Config, sshRunner *ssh.Runner) erro
 		return fmt.Errorf("Failed to remove Pull secret %w: %s", err, stderr)
 	}
 
-	cmdArgs = []string{"delete", "machineconfigs", "--all"}
-	_, stderr, err = ocConfig.RunOcCommand(cmdArgs...)
+	return nil
+}
+
+func RemoveOldRenderedMachineConfig(ocConfig oc.Config) error {
+	// This block (#L179-183) should be removed as soon as we start shipping with 4.8 bundle.
+	// This check is only make sure if there is any machineconfig resource or not because
+	// in 4.7 we disabled mco and also deleted all the machineconfig/machineconfig-pools.
+	// For 4.8 we don't disable mco and it does contain the machineconfigs.
+	stdout, stderr, err := ocConfig.RunOcCommand("get mc --sort-by=.metadata.creationTimestamp --no-headers -oname")
 	if err != nil {
-		return fmt.Errorf("Failed to remove machineconfigs %w: %s", err, stderr)
+		return fmt.Errorf("failed to get machineconfig resource %w: %s", err, stderr)
+	}
+	sortedMachineConfigsWithTime := strings.Split(stdout, "\n")
+	if len(sortedMachineConfigsWithTime) == 0 {
+		return nil
 	}
 
-	cmdArgs = []string{"delete", "machineconfigpools", "--all"}
-	_, stderr, err = ocConfig.RunOcCommand(cmdArgs...)
-	if err != nil {
-		return fmt.Errorf("Failed to remove machineconfigpools %w: %s", err, stderr)
+	var (
+		renderedMaster []string
```

Comment:
could be `renderedMasters` and `renderedWorkers` since an array can hold multiple masters/workers.

Reply:
In our case we have only single master/worker, config may be multiple.